src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
You mean, that the user must now this kind of internal logic docker requires?
It is bad design by dockers remote api, because it is not abstracted by the api.
It tooks me really a lot of time to figure out this issue. And it is not documented anywhere. The docker cli takes the same --link argument for new and old networks and handle it in the client.